gtsam/base/numericalDerivative.h

```diff
@@ -176,13 +169,14 @@ typename internal::FixedSizeMatrix<Y,X1>::type numericalDerivative21(const boost
       "Template argument Y must be a manifold type.");
   BOOST_STATIC_ASSERT_MSG( (boost::is_base_of<gtsam::manifold_tag, typename traits<X1>::structure_category>::value),
       "Template argument X1 must be a manifold type.");
-  return numericalDerivative11<Y, X1, N>(boost::bind(h, _1, boost::cref(x2)), x1, delta);
+  return numericalDerivative11<Y, X1, N>(boost::bind(h, boost::placeholders::_1, boost::cref(x2)), x1, delta);
 }
 
 /** use a raw C++ function pointer */
 template<class Y, class X1, class X2>
 typename internal::FixedSizeMatrix<Y,X1>::type numericalDerivative21(Y (*h)(const X1&, const X2&), const X1& x1,
     const X2& x2, double delta = 1e-5) {
+  using namespace boost::placeholders;
```

Comment:
The only variables in `boost::placeholders` are the `_X` where `X` goes from 1 to 9. [Link](https://github.com/boostorg/bind/blob/develop/include/boost/bind/placeholders.hpp)

As we use multiple `_X`, I thought it would more readable to use `boost::placeholders` instead of manually specifying each `boost::placeholders::_X`.

Reply:
I think both are fine, and I personally prefer `using xxx::_1` in rarely changed code, and `using ns` in scripts.

Reply:
@ProfFan can you clarify your preference concerning `using` in header files, specifically in an existing namespace scope vs a local function scope?

Reply:
@acxz yes I am aware. My recommendation is to use `using boost::placeholders::_1;` and `using boost::placeholders::_2;` etc at the top level of the namespace rather than including the full namespace via a `using namespace` statement.

Reply:
So then is this the same concern as the previous comment chain? If so can we close this and just keep the conversation to one comment chain?

If you want me to use `using` instead of `using namespace` I can go ahead and do that, but do note that I'll be doing that inside of every method. I would prefer we resolve the first concern first before.

Reply:
Sorry no, having a `using` inside every method is not a good idea for reasons I've mentioned before.

Reply:
OK, seems a whole argument developed here :-) I am in favor of explicitly using `boost::placeholders::_1` and 1`boost::placeholders::_2` in this file. The google style guide explicitly forbids importing namespaces, and it's a bad idea in header files, *even* inside a scope like above. The reason is (apparently) that some (microsoft!) compilers don't respect the scope.

Reply:
Second Frank here

Reply:
Okay, then we should be consistent since GTDynamics doesn't explicitly follow this. :)

Reply:
> some (microsoft!) compilers don't respect the scope

yikes that is truly unfortunate. I'll add a commit that explicitly uses `boost::placeholders::_X` in the header files.